### PR TITLE
Webops 693/exec with pipe

### PR DESCRIPTION
**Pull request by thepwagner**
Just dropping breadcrumb; will merge+flatten once dockerized-testing is stabled/merged.

### CODE REVIEW

File: bin/ssh-key [hunk 1]
```diff
@@ -2,34 +2,60 @@
 
 USERNAME=$1
 PUBKEY="$2 $3 $4"
+
+if [[ -z "${PUBKEY// }" ]]; then
+    exit 1
```
Comment (DeviaVir) [inline bin/ssh-key]: tabs or spaces?

File: bin/ssh-key [hunk 2]
```diff
@@ -2,34 +2,60 @@
...
+
+if [ $FOUND_KEY -eq 0 ]; then
+	NEW_KEYS="${PUBKEY_LINE}\n${NEW_KEYS}"
+fi
 
 if [ $SKIP_KEY -eq 0 ]; then
     echo "Adding key ${NEW_KEY_FP}..."
-    cat ${KEYFILE} >> .ssh/authorized_keys
+    echo -e $NEW_KEYS > .ssh/authorized_keys
     git-shell-commands/backup
```
Comment (DeviaVir) [inline bin/ssh-key]: spaces here, while the rest of the file use tabs (not your fault, but easy fix?)

File: bin/ssh-key (hunk 1 above)
Comment (lrvick) [inline bin/ssh-key]: bash should always be tabs. Bash is not aware of spaces, but it is aware of tabs and you can even tab-indent things like HEREDOCs without them showing up in the result.

Any time you see me using anything but tabs, It is wrong :-P

**Comment by lrvick** [comment]
+1, go for it whenever

**Comment by DeviaVir** [comment]
+1

**Pull request merged**